### DISCUSSION TITLE
PBCopySurvey ¿Como podría copiar un survey+survey?

### DISCUSSION THREAD

**Discussion by AlejandroLoraG** [Cotalker/documentation#417] [2. Questions and Answers]
### **Hola a tod@s!**

Estoy creando tareas en una maquina de estado desde un initial form, este survey contiene una question de tipo survey+survey dentro suyo. El cliente puede agregar cuantos survey+survey quiera, y al enviar el formulario principal (o padre) el survey trigger del initial form itera sobre cada uno de los survey+survey que contiene el formulario padre y crea una tarea, es decir, **el survey trigger crea una tarea por cada survey+survey que contenga el initial form.**

### El requerimiento es el siguiente:

- Deseo poder mandar al canal que se le crea a la tarea su survey+survey correspondiente, para esto ocupo PBCopySurvey.
- PBCopySurvey me pide:

    _**1. formId, corresponde al formulario que quiero copiar.
    2. channelId, corresponde al ID del canal donde quiero enviar el formulario copiado.
    3. channelId, corresponde al ID del canal donde existe el formulario que quiero copiar
    4. userId, corresponde al ID del usuario que envía el formulario.**_

El problema es que el formId de un survey+survey es el mismo que el de su padre, entonces al momento de intentar el formulario especifico para la tarea, se copia el padre con todos los survey+survey que contiene este.

**Rutina PBCopySurvey**
![image](https://user-images.githubusercontent.com/51095635/133677340-8e4e0e84-f3d2-46bd-be07-f87e0d56e09d.png)

**Contexto answer, correspondiente al formulario padre.**
![image](https://user-images.githubusercontent.com/51095635/133677582-a529fdaf-cbcb-4899-ac4d-e96d84e9b536.png)

**Contexto answer, correspondiente a survey+survey dentro del formulario padre**
![image](https://user-images.githubusercontent.com/51095635/133677684-1519bb73-daa3-4a6d-a5fe-4e4758e610d0.png)

Sería de mucha ayuda poder identificar de manera distinta a los survey+survey para poder acceder a el para implementaciones como esta.

Muchas muchas gracias! Son sec@s!
Saludos y pasenlo genial este fin de semana <3

**Comment by NicolasPLG** [comment]
Hola @AlejandroLoraG 

No entiendo bien la consulta :(, pero si quieres saber los distintos survey/survey tendrías que usar el bot iterar sobre las distintas answers y buscar el id del formulario y luego crear la tarea.

La otra opción es usar un bot global, relacionado al survey/survey y usando los PB bots.
get task from a channel
nwr answers
Create task (y colocar en PARENT CHANNEL el $VALUE#channel)
Wait de +/- 6000 a 10000 ms
Copy survey
send ack

No sé si te sirve esa config de rutina de lo contrario, confírmame si es lo que esperas de esta rutina, de lo contrario confírmame lo siguiente:
Quieres crear una task para cada survey/survey que se ingresa a la task padre?.

Atento =>

**Reply by AlejandroLoraG** [reply]
Hola holaa @NicolasPLG

Pucha, es un problema respecto a como poder usar PBCopySurvey para un survey+survey. La creacion de tasks y todo eso ya está implementado.

**Reply by NicolasPLG** [reply]
si quieres usar el s+s tendrás que hacer una nwr de answers para luego hacer una iteración en el campo del survey con un valor de la salida del survey y en la parte id del formulario colocas $VALUE#valor del iterate, y tendrías el valor del survey.

Creo que falta entremedio un nwr para identificar el survey, pero eso debería ser. Si quieres puedo hacer pruebas con esas rutinas para tener mejor clara mi respuesta.

Quedo atento =>
